### PR TITLE
react native integration tutorial

### PR DESCRIPTION
**Pull request by mateosilguero**
## Summary

Integrate Flipper with React Native, is a bit different than a native app. I wrote this tutorial inspired by @nparashuram 's blog. http://blog.nparashuram.com/2019/09/using-flipper-with-react-native.html

## Changelog

Flipper + React Native Documentation

### CODE REVIEW

**Comment by passy** [comment]
Omg, this is amazing! 🎉 Thank you so much. I'll give this a closer look later today.

File: docs/react-native.md [hunk 1]
```diff
@@ -0,0 +1,353 @@
+---
+id: react-native
+title: React Native
+---
+
+This tutorial was inspired by http://blog.nparashuram.com/2019/09/using-flipper-with-react-native.html
```
Comment (passy) [inline docs/react-native.md]: Maybe put this in italics to distinguish this a bit from the rest and make it "inspired by [a blog post by Ram N](http://blog.nparashuram.com/2019/09/using-flipper-with-react-native.html)"?

**Comment by passy** [comment]
Before I go into the details: This adds quite a bit of duplication. Do you think we could just add this as another paragraph on the existing "Getting Started" page underneath Android and iOS?

File: docs/react-native.md [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+      exclude group:'com.facebook.yoga'
+      exclude group:'com.facebook.flipper', module: 'fbjni'
+      exclude group:'com.facebook.litho', module: 'litho-annotations'
+      exclude group:'com.squareup.okhttp3'
+    }
+  }
+}
+```
+
+Now, we must create a new file inside `android/app/src/debug/java/com/yourappname/ReactNativeFlipper.java`. this a `debug`puorpose file.
```
Comment (passy) [inline docs/react-native.md]: Not quite sure what the second sentence here meant to say.

File: docs/react-native.md [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+- Shared Preferences
+- Crash Reporter
+- React devtools
+
+```java
+/\*\*
+- Copyright (c) Facebook, Inc. and its affiliates.
+- <p>This source code is licensed under the MIT license found in the LICENSE file in the root
+- directory of this source tree.
+\*/
```
Comment (passy) [inline docs/react-native.md]: Some syntax problems here, but it can generally be removed.

File: docs/react-native.md [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+  @Override
+  public void onCreate() {
+    ...
+    initializeFlipper(this, getReactNativeHost().getReactInstanceManager());
+  }
+
+  /**
+    - Loads Flipper in React Native templates.
+    - @param context
+  \*/
```
Comment (passy) [inline docs/react-native.md]: Some escaping stuff happening again.

File: docs/react-native.md [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+        e.printStackTrace();
+      } catch (InvocationTargetException e) {
+        e.printStackTrace();
+      }
+    }
+  }
+}
+```
+
+Finally, open Flipper desktop app, and run `npm run android` in your terminal.
```
Comment (passy) [inline docs/react-native.md]: *the Flipper desktop app.

Comment (passy) [reply docs/react-native.md]: We also generally advise using `yarn`.

**Comment by mateosilguero** [comment]
> Before I go into the details: This adds quite a bit of duplication. Do you think we could just add this as another paragraph on the existing "Getting Started" page underneath Android and iOS?

Like
Getting Started
  - setup android
  - setup ios
  - setup RN

?

**Comment by mateosilguero** [comment]
> Some comments for the RN-specific bits. :)

thanks for feedback ! will push some changes in a while.

**Comment by passy** [comment]
> > Before I go into the details: This adds quite a bit of duplication. Do you think we could just add this as another paragraph on the existing "Getting Started" page underneath Android and iOS?
> 
> Like
> Getting Started
> 
> * setup android
> * setup ios
> * setup RN
> 
> ?

Yes, exactly! Otherwise we'll have to update two pages whenever we want to make changes to the overlapping material which isn't ideal. It's also not great from a SEO perspective.

**Comment by mateosilguero** [comment]
Done :)

**Comment by priteshrnandgaonkar** [comment]
It would be great if we have a sample app for RN too, so that users can directly run it and see what features are provided by Flipper for RN. Usually we prefer having Sample apps, for all the different platforms and technologies that we support. We can have that as a separate PR.

**Comment by passy** [comment]
@priteshrnandgaonkar I've got a super basic Android template just adding the dependencies and setting up an RN view, but because of our current `fbjni` incompatibilities, I can't land it.

**Comment by priteshrnandgaonkar** [comment]
@passy We can have an iOS example to begin with.

**Comment by passy** [comment]
> @passy We can have an iOS example to begin with.

That would be great!

**Comment by mateosilguero** [comment]
Great, I'm working on a RN example

**Comment by passy** [comment]
Sorry, I won't have the time to review this today, but I'll give it another look tomorrow. :)

File: docs/getting-started.md [hunk 6]
```diff
@@ -298,8 +306,333 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+    exclude group:'com.facebook.flipper', module: 'fbjni'
+    exclude group:'com.facebook.litho', module: 'litho-annotations'
+    exclude group:'com.squareup.okhttp3'
+  }
+}
+```
+
+Now, we must create a new file inside `android/app/src/debug/java/com/yourappname/ReactNativeFlipper.java`.
+
+Here, there are the following plugins integrations:
```
Comment (passy) [inline docs/getting-started.md]:
```suggestion
These are the suggested plugins integrations:
```

File: docs/getting-started.md [hunk 7]
```diff
@@ -298,8 +306,333 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+- Crash Reporter
+- React devtools
+
+```java
+/**
+* Copyright (c) Facebook, Inc. and its affiliates.
+*
+* <p>This source code is licensed under the MIT license found in the LICENSE file in the root
+* directory of this source tree.
+*/
```
Comment (passy) [inline docs/getting-started.md]: No need to include this. Especially since this won't apply to users' apps.

File: docs/getting-started.md [hunk 8]
```diff
@@ -298,8 +306,333 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+import com.facebook.flipper.plugins.sharedpreferences.SharedPreferencesFlipperPlugin;
+import com.facebook.react.ReactInstanceManager;
+import com.facebook.react.bridge.ReactContext;
+import com.facebook.react.modules.network.NetworkingModule;
+import okhttp3.OkHttpClient;
+
+public class ReactNativeFlipper {
+
+  public static void initializeFlipper(Context context, final ReactInstanceManager reactInstanceManager) {
+    if (FlipperUtils.shouldEnableFlipper(context)) {
```
Comment (passy) [inline docs/getting-started.md]: Can you turn this into an early return please?

File: docs/getting-started.md [hunk 9]
```diff
@@ -298,8 +306,333 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+      }
+    }
+  }
+}
+```
+
+Now you can initialize Flipper in your Application's `onCreate` method, which involves
+initializing SoLoader (for loading the C++ part of Flipper) and starting a `FlipperClient`.
+
+For this, we should edit our `android/app/src/main/java/com/yourappname/MainApplication.java` file.
```
Comment (passy) [inline docs/getting-started.md]:
```suggestion
For this, we edit the `android/app/src/main/java/com/yourappname/MainApplication.java` file.
```

File: docs/getting-started.md [hunk 10]
```diff
@@ -298,8 +306,333 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+<div class="warning">
+
+On the first run of `pod install`, `FB_SONARKIT_ENABLED=1` may not be added in the "Build Settings" of your project, but in all the subsequent runs of `pod install`, the above `post_install` hook successfully adds the compiler flag. So before running your app, make sure that `FB_SONARKIT_ENABLED=1` is present in `OTHER_CFLAGS` and `OTHER_SWIFT_FLAGS` for Objective-C and Swift projects respectively.
+
+</div>
+
+Install the dependencies by running `cd ios && pod install`. You can now import and initialize Flipper in your
+`ios/your-app-name/AppDelegate.m`.
+
+Here, there are the following plugins integrations:
```
Comment (passy) [inline docs/getting-started.md]:
```suggestion
The code below enabled the following integrations:
```

**Comment by mateosilguero** [comment]
Done ! changes was commited, and i made a new PR (#574) with the examples folder.

**Comment by passy** [comment]
Thanks for making all those changes! 🥳

**Pull request closed without merging**